### PR TITLE
Login UI Changes

### PR DESCRIPTION
**Pull request by cf2080**
# Issue/Ticket

CLOSES #XX
# Description
Added eventbus file in services which aided in login button not appearing.
Also added code in navbar.vue, login.vue, and main.js file

## Type of change

List one of the following: Chore

# How Has This Been Tested?

# Notes

Put any notes here

### CODE REVIEW

File: Frontend/src/components/NavBar.vue [hunk 1]
```diff
@@ -11,13 +11,33 @@
     <v-btn to="add" flat>Register</v-btn>
     <v-btn to="key" flat>Generate</v-btn>
     <v-btn to="delete" flat>Delete</v-btn>
-    <v-btn to="login" flat>Login</v-btn>
+    <v-btn to="login" flat v-if="isLogged === false">Login</v-btn>
```
Comment (alfredovargasjr) [inline Frontend/src/components/NavBar.vue]: IsLogged should return true or not already

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Frontend/src/views/Login.vue [hunk 2]
```diff
@@ -31,8 +31,9 @@
                     password: this.$data.password
                })
                .then(resp => {
-                   localStorage.email = this.email;
-                   localStorage.token = resp.data; 
+                   let respData = resp.data
+                   localStorage.setItem('token', respData)
```
Comment (alfredovargasjr) [inline Frontend/src/views/Login.vue:66]: store the email in local as well, update password needs it

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Frontend/src/views/Login.vue (hunk 2 above)
Comment (julianpoy) [reply Frontend/src/views/Login.vue:66]: Email should not be stored in localStorage. This should be fetched from the server.

File: Frontend/src/views/Login.vue [hunk 3]
```diff
@@ -31,8 +31,10 @@
                     password: this.$data.password
                })
                .then(resp => {
-                   localStorage.email = this.email;
-                   localStorage.token = resp.data; 
+                   let respData = resp.data
+                   localStorage.setItem('email', this.$data.email)
+                   localStorage.setItem('token', respData)
+                   this.$bus.$emit('logged', 'User logged')
```
Comment (julianpoy) [inline Frontend/src/views/Login.vue]: You should not need an eventbus for this feature.

Comment (cf2080) [reply Frontend/src/views/Login.vue]: I did do it at first without using an event bus but you had to refresh the page in order to see the change.

Comment (julianpoy) [reply Frontend/src/views/Login.vue]: It's possible to do this without event bus. Just storr an isLoggedIn variable in a shared service.

File: Frontend/src/views/Login.vue (hunk 2 above)
Comment (alfredovargasjr) [reply Frontend/src/views/Login.vue:66]: @julianpoy reason for that extra step?

File: Frontend/src/views/Login.vue (hunk 3 above)
Comment (cf2080) [reply Frontend/src/views/Login.vue]: so like using Vuex? I believe Vuex works as a shared service.

File: Frontend/src/main.js [hunk 4]
```diff
@@ -2,9 +2,13 @@ import Vue from 'vue'
 import './plugins/vuetify'
 import App from './App.vue'
 import router from './router'
+import EventBus from './services/EventBus'
+
 
 Vue.config.productionTip = false
 
+Vue.prototype.$bus = EventBus
```
Comment (julianpoy) [inline Frontend/src/main.js]: Why do we have event bus in here still?

File: Frontend/src/views/Login.vue (hunk 2 above)
Comment (julianpoy) [reply Frontend/src/views/Login.vue:66]: Not good practice - localStorage should not be used to store personal/identifying data.

Also not RESTful.

File: Frontend/src/views/Login.vue (hunk 3 above)
Comment (julianpoy) [reply Frontend/src/views/Login.vue]: No need for vuex either. Localstorage should do everything you need concerning state.

**Review by julianpoy — DISMISSED** [review]
See additional comments

File: Frontend/src/components/NavBar.vue [hunk 5]
```diff
@@ -11,13 +11,50 @@
     <v-btn to="add" flat>App Register</v-btn>
     <v-btn to="key" flat>Generate Key</v-btn>
     <v-btn to="delete" flat>App Delete</v-btn>
-    <v-btn to="login" flat>Login</v-btn>
+    <v-btn to="login" flat v-if="!isLoggedIn.isLogin">Login</v-btn>
```
Comment (alfredovargasjr) [inline Frontend/src/components/NavBar.vue:11]: if the page is refreshed, then the login button appears again. and register still remains when i register

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]
Get changes from master

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]
Delete those two files made in the DAL, those were due to my mistake

**Review by alfredovargasjr — APPROVED** [review]

**Pull request merged**